src/shared_gui_components/BuildingComponentDialogCentralWidget.cpp

```diff
@@ -169,6 +168,34 @@ void BuildingComponentDialogCentralWidget::setTid() {
   requestComponents(m_filterType, m_tid, m_pageIdx, m_searchString);
 }
 
+std::vector<openstudio::BCLSearchResult> BuildingComponentDialogCentralWidget::fetchAndSortResponses(const std::string& filterType, int tid, const QString& searchString) {
+  m_allResponses.clear();
+  RemoteBCL remoteBCL;
```

Comment:
You can set `RemoteBCL::setResultsPerQuery` to 100 which might make this go faster:

https://github.com/NREL/OpenStudio/blob/develop/src/utilities/bcl/RemoteBCL.cpp#L950

Reply:
Did you try this @antonszilasi ?

Reply:
Im not sure how to measure this?

Reply:
If it works, then it's almost certainly faster to do say 2 queries with 100 results per query than 20 queries with 10 results per query.  I'd just set the value to 100, set a break point to check that it is actually using that value for number of results, and then test to make sure it still works.